### ISSUE TITLE
sign bug

### ISSUE THREAD

**Issue by AStupidBear** [JuliaComputing/ArrayFire.jl#31]
```julia
julia> ju=[0.5,-0.5];

julia> af=AFArray(ju);

julia> sign(ju)
2-element Array{Float64,1}:
  1.0
 -1.0

julia> sign(af)
2-element ArrayFire.AFArray{Float64,1}:
 0.0
 1.0

 # if this is due to the definition of sign in ArrayFire, I cannot rescue
 # this problem
 julia> -af
2-element ArrayFire.AFArray{Float64,1}:
  0.5
 -0.5

julia> 0-af
2-element ArrayFire.AFArray{Float64,1}:
  0.5
 -0.5

julia> 2*sign(af)-1
2-element ArrayFire.AFArray{Float64,1}:
 -1.0
  1.0

julia> 1-2*sign(af)
2-element ArrayFire.AFArray{Float64,1}:
 -1.0
  1.0
```

Those are fatal to my project !
I think it's time to write test for all functions.

**Comment by ranjanan** [comment]
@AStupidBear The `sign` function in ArrayFire simply uses a different convention as compared to Base Julia.

The `-af` bit is a bug. Pushing a fix now.

**Comment by ranjanan** [comment]
Fixed by e2122bf2af9a2695864bd3be8ccb4f557f96496f

**Comment by ranjanan** [comment]
Also wrote tests to reflect the fixes.
